.azure-pipelines/templates/install-go.yml

```diff
@@ -0,0 +1,19 @@
+parameters:
+  packages: ''
+  version: '1.12.1'
+
+steps:
+- task: GoTool@0
+  inputs:
+    version: ${{ parameters.version }}
+  displayName: 'Install go'
+- powershell: |
+    $env:GOBIN = "$(GOPATH)\bin"
+    echo "##vso[task.setvariable variable=GOBIN]$env:GOBIN"
+    go env
+    $env:PATH += ";$env:GOBIN"
+    echo "##vso[task.setvariable variable=PATH]$env:PATH"
+    echo $env:PATH
+- ${{ if ne(parameters.packages, '') }}:
+  - powershell: go get -u ${{ parameters.packages }}
```

Comment:
Do this in a different directory from the agent code so that it doesn't try and download everything from the go.mod file again.  Also make sure the GO111MODULE=on envvar is set here.

Reply:
`GO111MODULE=on` is already set as a global env var.
Changing to a different directory now installs the packages in seconds.